dataframe-csv/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/Compression.kt

```diff
@@ -0,0 +1,105 @@
+package org.jetbrains.kotlinx.dataframe.io
+
+import java.io.File
+import java.io.InputStream
+import java.net.URL
+import java.nio.file.Path
+import java.util.zip.GZIPInputStream
+import java.util.zip.InflaterInputStream
+import java.util.zip.ZipInputStream
+
+/**
+ * Compression algorithm to use when reading csv files.
+ * We support [GZIP][Compression.Gzip] and [ZIP][Compression.Zip] compression out of the box.
+ *
+ * Custom decompression algorithms can be added by creating an instance of [Custom].
+ *
+ * @param wrapStream function that wraps any [InputStream] into a decompressing [InflaterInputStream] stream
+ */
+public sealed class Compression<I : InputStream>(public open val wrapStream: (InputStream) -> I) {
```

Comment:
Is this type parameter needed? In the end readCsv has Compression<*>, so it's the same as `Compression(public open val wrapStream: (InputStream) -> InputStream)`